core/challenges.go

```diff
@@ -5,31 +5,73 @@
 
 package core
 
+import (
+	"encoding/json"
+	"github.com/letsencrypt/boulder/Godeps/_workspace/src/github.com/letsencrypt/go-jose"
+)
+
+func newChallenge(challengeType string, accountKey *jose.JsonWebKey) (Challenge, error) {
+	ak := AuthorizedKey{
+		Token: NewToken(),
+		Key:   accountKey,
+	}
+
+	jsonAK, err := json.Marshal(ak)
+	if err != nil {
+		return Challenge{}, err
+	}
+
+	return Challenge{
+		Type:          challengeType,
+		Status:        StatusPending,
+		AccountKey:    accountKey,
+		AuthorizedKey: jsonAK,
+	}, nil
+}
+
+//----- BEGIN TO DELETE -----
 // SimpleHTTPChallenge constructs a random HTTP challenge
-func SimpleHTTPChallenge() Challenge {
+func SimpleHTTPChallenge(accountKey *jose.JsonWebKey) (Challenge, error) {
```

Comment:
You didn't remove these errors nor leave a comment on my thread saying why you didn't.